File: src/runtime/js/std/os_native.js
// Generated std/os.js delegates host bindings through this native seam.

import { mkdirSync } from "node:fs";
import { resolve } from "node:path";

export function getcwd() {
    return process.cwd();
}

export function mkdir(p) {
    mkdirSync(p);
}

export function makedirs(p, exist_ok = false) {
    mkdirSync(p, { recursive: exist_ok });
}

File: src/runtime/julia/std/os_native.jl
# Native implementation for pytra.std.os
# Generated std/os.jl delegates host bindings through this native seam.

module __OsNative
    function getcwd()
        return pwd()
    end
    function mkdir(path, exist_ok)
        if exist_ok && isdir(path)
            return nothing
        end
        Base.mkdir(path)
        return nothing
    end
    function makedirs(path, exist_ok)
        if exist_ok && isdir(path)
            return nothing
        end
        Base.mkpath(path)
        return nothing
    end
end
